lib/vf_measurement/src/lib.rs

```diff
@@ -40,22 +42,26 @@ impl<'a> QuantityValue {
     }
 }
 
-pub fn add(q1: QuantityValue, q2: QuantityValue) -> QuantityValue {
+pub fn add(q1: QuantityValue, q2: QuantityValue) -> RecordAPIResult<QuantityValue> {
     if q1.has_unit != q2.has_unit {
-        panic!("Unimplemented! Need to enable unit conversions in QuantityValue math");
-    }
-    QuantityValue {
-        has_numerical_value: q1.has_numerical_value + q2.has_numerical_value,
-        has_unit: q1.has_unit,
+        return Err(DataIntegrityError::Wasm(wasm_error!(WasmErrorInner::Guest(format!("Mismatching unit types. Trying to add {:?} with {:?}", q1.get_unit(), q2.get_unit()))).into()));
```

Comment:
another option here to improve on the resulting error formatting in the client, is to add a specific error variant to DataIntegrityError. Looking at the variants, you can get a sense how it interpolates variables into the error message. I'd like to see this one thing changed.